Research/lva-node-app/app.js

```diff
@@ -0,0 +1,337 @@
+* @returns {void}
+*/
+app.get('/getglobals', function (req, res) 
+{
+  console.log("get globals get function!");
+  let globals = 
+  [
+    {
+      "name": "graphInstances",
+      "value": graphInstances
+    },
+    {
+      "name": "graphTopologies",
+      "value": graphTopologies
+    },
+    {
+      "name": "cameras",
+      "value": cameras
+    }
+  ];
+  res.send(JSON.stringify(globals));
+})
+
+/**
+* start sending live stream hub messages to client via websocket 
+* @returns {void}
+*/
+app.get('/hubMessages', function(req, res)
+{
+  if (!validCredentialsAreSet(res))
+  {
+    return;
+  }
+  else
+  {
+    receiveHubMessages();
+    //always close requests
+    res.send();
+  }  
+})
+
+/**
+ * stop receiving messages from iot hub
+ * @returns {void}
+ */
+app.get('/stopMessages', function(req, res)
+{
+  eventHubReader.stopReadMessage();
+  res.send();
+})
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// handle POST requests
+
+/**
+ * invoke LVA methods handler
+ * @returns {void}
+ */
+app.post('/runmethod', function (req, res) 
+{ 
+    if (!validCredentialsAreSet(res))
+    {
+      return;
+    }
+    let methodName=req.body.methodName;
+    invokeLVAMethod(req, res).then(response => 
+    {
+      let obj=[{method: methodName}, response.result];
+      //send results of invoking the method back to the client
+      res.send(obj); 
+    }).catch(error =>
+    {
+      console.error(error.message);
+      res.status(400).send(error.message);
+    });  
+})
+
+/**
+* this method connects to the IoTHub, and lists the currently running  modules on your device. Ensures connection works properly!
+* Here be dragons :) I double-promise
+*/
+app.post('/connectToIotHub', function (req, res) 
+{
+    setConfigs(req, res);
+    iotHubConnection(req, res).then(response => { //.then is promise resolved callback. must do here because invokeLVAMethod is async
+      let modules="";
+        for(let i=0; i<response.responseBody.length; i++)
+        {
+          modules+=response.responseBody[i].moduleId+" ";
+          console.log(response.responseBody[i].moduleId);
+        }
+        res.send("Successfully connected to your IoTHub. Found the following modules on your device: "+modules);
+    }, reason =>  //on error send error response
+    {
+      console.log(reason);
+      res.send("Unable to connect to device. Error response: "+ JSON.parse(reason.responseBody).Message);
+    });          
+}) 
+
+
+/**
+* update global variables when sent by user
+*/
+app.post('/globals', function (req, res)
+{
+  setGlobals(req, res);
+  res.send();
+})
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// defined functions
+
+/**
+* set global variables when passed by user 
+*/
+function setGlobals(req, res)
```

Comment:
why is set globals in a function but get globals (and most of the rest) aren't?

Reply:
Good point! I put it inside the app.post. Thanks for catching this